File: Book/Fortran/Fig_12.18_ompParVecPi.f90
          PROGRAM MAIN

          USE OMP_LIB
          IMPLICIT NONE

          INTEGER :: i

          INTEGER, PARAMETER :: num_steps = 100000
          REAL :: x, pi, sum, step

          sum = 0.0

          step = 1.0/num_steps

!$omp parallel do simd private(x) reduction(+:sum)
          DO i = 1, num_steps
             x = (i - 0.5) * step
             sum = sum + 4.0 / (1.0 + x * x)
          ENDDO
!$omp end parallel do simd

          pi = step * sum

          print *, "pi=", pi
          END PROGRAM MAIN 

File: Book/C/Fig_12.18_ompParVecPi.c
#include<stdio.h>
#include <omp.h>
static long num_steps = 100000000;
double step;
int main ()
{
   int i;
   double x, pi, sum = 0.0;

   step = 1.0f / (double) num_steps;

   #pragma omp parallel for simd private(x) reduction(+:sum)
   for (i = 0; i < num_steps; i++) {
      x = (i + 0.5f) * step;
      sum += 4.0f / (1.0f + x * x);
   }

   pi = step * sum;
   printf("pi=%f\n",pi);
}	  
